webpack/JobWizard/steps/AdvancedFields/DescriptionField.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormGroup, TextInput, Button } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+export const DescriptionField = ({ inputs, value, setValue }) => {
+  const generateDesc = () => {
+    let newDesc = value;
+    if (value) {
+      const re = new RegExp('%\\{([^\\}]+)\\}', 'gm');
+      const results = [...newDesc.matchAll(re)].map(result => ({
+        name: result[1],
+        text: result[0],
+      }));
+      results.forEach(result => {
+        newDesc = newDesc.replace(
+          result.text,
+          // TODO: Replace with the value of the input from Target Hosts step
+          inputs.find(input => input.name === result.name)?.name || result.text
+        );
+      });
+    }
+    return newDesc;
+  };
+  const [generatedDesc, setGeneratedDesc] = useState(generateDesc());
+  const [isPreview, setIsPreview] = useState(true);
+
+  const togglePreview = newValue => {
+    setGeneratedDesc(generateDesc());
```

Comment:
So the description will be generated when the user finished editing the template for the description

Reply:
should it be attached to the input onChange event than? or a `useEffect` on value changes?
btw, seem that the initial value regex worked, but when the value changes the regex doesn't seem to work:
![Run job (1)](https://user-images.githubusercontent.com/26363699/125443261-55298434-a004-4c77-b4c5-dd569aba6a15.gif)

Reply:
Use effect on value change is a too much updates as the user wont see the generated input before they toggle.
That is by design, because `test` is not a template input.
Here there are 2 user templates `command` and `task searc`:
![screencast-192 168 122 254-2021 07 13-13_31_41](https://user-images.githubusercontent.com/30431079/125445006-df33e83e-df29-4a31-85e5-769de737657b.gif)
The old form has the same functionality except it does replace `command` with the user input so at first the generated input would be `command ` with a blank as there is no user input at first.

Reply:
It depends on the template. I have a template where I added `task searc` as a user input.
You can create templates here `/job_templates/new`

Reply:
thanks, got that now, ignore my comment :)